pkg/porter/version.go

```diff
@@ -16,3 +26,55 @@ func (p *Porter) PrintVersion(opts version.Options) error {
 	}
 	return version.PrintVersion(p.Context, opts, metadata)
 }
+
+func getSystemInfo() *SystemInfo {
+	return &SystemInfo{
+		OS:            runtime.GOOS,
+		Arch:          runtime.GOARCH,
+	}
+}
+
+func (system *SystemInfo) printSystemInfo(out io.Writer) error {
+	_, err := fmt.Fprintf(out, "os: %s\n", system.OS)
+	_, err = fmt.Fprintf(out, "arch: %s\n", system.Arch)
+	return err
+}
+
+func printSectionHeader(out io.Writer, header string){
+	underline := "-------"
+	_, _ = fmt.Fprintf(out, "\n%s\n%s\n", header, underline)
+
+}
+
+func (p *Porter) PrintDebugInfo(opts version.Options) error {
+	// force opts to print version as plaintext
+	opts.RawFormat = string(printer.FormatPlaintext)
+	err := p.PrintVersion(opts)
+	if err != nil {
+		return errors.Wrap(err, "Failed to print version")
+	}
+
+	printSectionHeader(p.Context.Out, "System")
```

Comment:
You can achieve this functionality using go templates. I think it will more readable. Just a opinion

**example:**
```golang
const ts = `
version: {{.Version}}
{{if .Sys -}}

System
--------
os: {{.Sys.OS}}
arch: {{.Sys.Arch}}
{{- end}}
`
v := Version{
	Version: "1.0.0",
	Sys:     nil,
}
tmpl, _ := template.New("system").Parse(ts)
tmpl.Execute(p.Context.Out, v)
```